pkgs/sops-install-secrets/main.go

```diff
@@ -102,6 +103,30 @@ type appContext struct {
 	checkMode   CheckMode
 }
 
+func secureSymlinkChown(symlinkToCheck, expectedTarget string, owner, group int) error {
+	fd, err := unix.Open(symlinkToCheck, unix.O_CLOEXEC|unix.O_PATH|unix.O_NOFOLLOW, 0)
+	if err != nil {
+			return fmt.Errorf("Failed to open %s: %w", symlinkToCheck, err)
+	}
+	defer unix.Close(fd)
+
+	buf := make([]byte, len(expectedTarget) + 1)
+	n, err := unix.Readlinkat(fd, "", buf)
+	if err != nil {
+		return fmt.Errorf("couldn't readlinkat %s", symlinkToCheck)
+	}
+	actualTarget := string(buf[:n]) // chop off null byte
```

Comment:
According to `readlink(2)` there is no terminating null byte appended.

Reply:
I'll look into it, but without it there was a \x00 that was causing the string comparison to fail.

Reply:
There's a relevant note at the bottom: https://www.gnu.org/software/gnulib/manual/html_node/readlinkat.html

Reply:
The `\x00` is because your byte buffer is 1 byte longer than your string. You need to compare `buf[:n]` to your string.

Reply:
never mind, I think I understand from the other thread.

Reply:
so your code is actually correct but maybe fix the comment because it is misleading. In fact you could cast the input string to a byte slice as well and compare those to make it very explicit that you are just comparing bytes.